src/main/java/com/connexta/store/adaptors/StorageAdaptor.java

```diff
@@ -39,4 +39,12 @@ void store(
    */
   @NotNull
   StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key) throws RetrieveException;
+
+  /**
+   * Updates the {@link StoreStatus} of {@code dataset} for a given {@code datasetId}
+   *
+   * @param datasetId the Id of a dataset
+   * @param storeStatus the new {@link StoreStatus} for a dataset
+   */
+  void updateStatus(String datasetId, String storeStatus);
```

Comment:
I like the direction of this. I think StoreStatus can go away and maybe just pass in a parameter called `String group`, and let the client group Datasets arbitrarily.

Reply:
I made that class to make tagging more convenient and less error prone, not necessarily to restrict what a client can tag.

Reply:
I guess I'm getting at I think these constants belong only to the Store, not the Adapter

Reply:
Hmm, at that point should we just change this method to be `updateTag(String datasetId, String key, String value)` or a Map instead of one k-v pair.